cli/manage.go

```diff
@@ -128,34 +130,56 @@ func setupReplication(c *cli.Context, cluster cluster.Cluster, server *api.Serve
 	primary := api.NewPrimary(cluster, tlsConfig, &statusHandler{cluster, candidate, follower}, c.Bool("cors"))
 	replica := api.NewReplica(primary, tlsConfig)
 
-	go func() {
-		candidate.RunForElection()
-		electedCh := candidate.ElectedCh()
-		for isElected := range electedCh {
+	go run(candidate, server, primary, replica)
+	go follow(follower, replica, addr)
+
+	server.SetHandler(primary)
+}
+
+func run(candidate *leadership.Candidate, server *api.Server, primary *mux.Router, replica *api.Replica) {
+	electedCh, errCh := candidate.RunForElection()
```

Comment:
They are `defer closed` in the Leadership side. Can't close them here as those are receive-only channels.